http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/websocket/WebSocketClientTestCase.java

```diff
@@ -50,20 +50,18 @@
     private HttpWsConnectorFactoryImpl httpConnectorFactory = new HttpWsConnectorFactoryImpl();
     private final String url = "ws://localhost:8490/websocket";
     private final int threadSleepTime = 100;
+    private Map<String, Object> senderProperties = new HashMap<>();
     private WebSocketClientConnector clientConnector;
-    private WebSocketRemoteServer remoteServer = new WebSocketRemoteServer(8490);
+    private WebSocketRemoteServer remoteServer = new WebSocketRemoteServer(8490, "xml, json");
 
     @BeforeClass
     public void setup() throws InterruptedException, ClientConnectorException {
         remoteServer.run();
-        Map<String, Object> senderProperties = new HashMap<>();
-
         senderProperties.put(Constants.REMOTE_ADDRESS, url);
-        senderProperties.put(Constants.WEBSOCKET_SUBPROTOCOLS, null);
         clientConnector = httpConnectorFactory.createWsClientConnector(senderProperties);
     }
 
-    @Test(description = "Test the WebSocket handshake and sending and receiving text messages.")
+    @Test(priority = 1, description = "Test the WebSocket handshake and sending and receiving text messages.")
```

Comment:
Is there any specific reason to use "priority" here? AFAIK, we have not used priority annotation in other places. Is there any dependencies in these test cases?

Reply:
Yes. We need this here since these tests should sequentially execute otherwise it can lead to intermittent test case failures.

Reply:
What is the root cause of intermittant test failures? What happen if they are executed in a real ballerina program in random order? If these tests does not have a dependency, I think we should not put priority to just get rid of an intermittant test failure because these sequence of events can occur in real programs. I would suggest to fix the intermittant failure than forcing an ordered execution.

Reply:
It is actually because of the class level variable "senderProperties" dependency in each test case. Those parameters are changing from one test case to another. If the test testSubProtocolNegotiation runs before others it can cause test failures only and only because of the class level "senderProperties" map that we are using.